packages/rocketchat-ui/client/views/app/room.js

```diff
@@ -878,6 +878,38 @@ Template.room.events({
 			Meteor.call('sendMessage', msgObject);
 		});
 	},
+	async 'click .pdf-open-in-modal'(event) {
+		event.preventDefault();
+		document.body.style.cursor = 'wait';
+		const pdfBlock = $(event.target).parent();
+		const pdfBlockFreeze = 'pdf-block-freeze';
+		pdfBlock.before(`<div class="${ pdfBlockFreeze }"></div>`);
+		const isElectron = (navigator.userAgent.toLowerCase().indexOf(' electron/') > -1);
+		const pdfLink = this._arguments[1].attachments[0].title_link;
+		const pdfId = `${ pdfLink.split('/')[2] }.pdf`;
+		if (isElectron) {
+			const newWindow = window.open(Meteor.absoluteUrl().replace(/\/$/, '') + pdfLink, pdfId, 'width=1,height=1');
```

Comment:
This line just initiates an event in the electron application, where you can send data and open it in a new window with pdf support (in electron app).